fix(transformer): TS transform handle when type exports first

overlookmotel opened the pull request:
Fix 2 bugs in TS transform:

1. Handle where export declaration precedes its import/definition. e.g. `export { X }; import { type X } from "x";`
2. Don't insert `export {}` statement if any other `export` statements remain.

Also refactor to simplify logic for removing imports/exports.

overlookmotel commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3833?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3833** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3833?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* **#3832** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3832?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#3831** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3831?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @overlookmotel and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
@Boshen Do you know why tests and conformance tasks have been skipped? I assume related to Graphite CI action. It's useful to see that these pass on every PR in a stack while working on it.

Dunqing commented:
> @Boshen Do you know why tests and conformance tasks have been skipped? I assume related to Graphite CI action. It's useful to see that these pass on every PR in a stack while working on it.

https://github.com/oxc-project/oxc/actions/runs/9629552106/job/26564917212

Dunqing reviewed (approved):
The way it's handled now seems clearer and I'll send another PR to remove the useless code. See #3827

overlookmotel committed:
fix(transformer): TS transform handle when type exports first (#3833)

Fix 2 bugs in TS transform:

1. Handle where export declaration precedes its import/definition. e.g. `export { X }; import { type X } from "x";`
2. Don't insert `export {}` statement if any other `export` statements remain.

Also refactor to simplify logic for removing imports/exports.
